phy/cluster/manual/gui.py

```diff
@@ -0,0 +1,381 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+"""GUI creator."""
+
+#------------------------------------------------------------------------------
+# Imports
+#------------------------------------------------------------------------------
+
+from ...utils.dock import DockWindow, _create_web_view, _title
+from ...utils import EventEmitter, debug
+from ...plot.view_models import BaseViewModel
+
+
+#------------------------------------------------------------------------------
+# Manual clustering window
+#------------------------------------------------------------------------------
+
+class KlustaViewa(EventEmitter):
+    """Manual clustering GUI.
+
+    This object represents a main window with:
+
+    * multiple views
+    * a wizard panel
+    * high-level clustering methods
+    * global keyboard shortcuts
+
+    """
+
+    def __init__(self, session, config=None):
+        super(KlustaViewa, self).__init__()
+        self.session = session
+        self.start()
+        self._dock = DockWindow(title=self.title)
+        self._load_config(config)
+        # Save the geometry state
+        self._create_gui_actions()
+        self._set_default_view_connections()
+
+    def _load_config(self, config=None):
+        """Load a GUI configuration dictionary."""
+        if config is None:
+            config = self.session.settings['gui_config']
+        for name, kwargs in config:
+            # GUI-specific keyword arguments position, size, maximized
+            position = kwargs.pop('position', None)
+            if name == 'wizard':
+                item = self._create_wizard_panel()
+            else:
+                clusters = self._cluster_ids
+                item = self.session.view_creator.add(name,
+                                                     cluster_ids=clusters,
+                                                     **kwargs)
+
+            self.add_view(item, title=name.title(), position=position)
+
+        # Load geometry state
+        gs = self.session.settings.get('gui_state', None)
+        if gs:
+            self._dock.restore_geometry_state(gs)
+
+    def show(self):
+        """Show the GUI"""
+        self._dock.show()
+
+    def start(self):
+        """Start the wizard."""
+        self.session.wizard.start()
+        self._cluster_ids = self.session.wizard.selection
+
+    @property
+    def main_window(self):
+        """Dock main window."""
+        return self._dock
+
+    @property
+    def title(self):
+        """Title of the main window."""
+        name = self.__class__.__name__
+        filename = self.session.model.kwik_path
+        clustering = self.session.model.clustering
+        channel_group = self.session.model.channel_group
+        template = "{name} - {filename} (shank {channel_group}, {clustering})"
+        return template.format(name=name,
```

Comment:
nicer to have `{filename} (shank {channel_group}, {clustering}) - {name}` to distinguish between multiple windows etc. the icon will take care of the `{name}` bit. cf: Google Chrome, Sublime, pretty much every app ever